src/clean_vhi_data.py

```diff
@@ -0,0 +1,282 @@
+    VHP.G04.C07.NJ.P1996027.VH.nc
+    """
+    substr = netcdf_filepath.split('/')[-1].split('.P')[0]
+    if subset:
+        assert subset_name!=None, "If you have set subset=True then you need to assign a subset name"
+        new_filename = f"STAR_{substr}_{t.year}_{t.month}_{t.day}_kenya_VH.nc"
+    else:
+        new_filename = f"STAR_{substr}_{t.year}_{t.month}_{t.day}_VH.nc"
+    return new_filename
+
+
+# ------------------------------------------------------------------------------
+# 1. Chop out East Africa
+# ------------------------------------------------------------------------------
+# from src.api_helpers import Region
+#
+# # kenya_region = Region(
+# #     name='kenya',
+# #     lonmin=33.501,
+# #     lonmax=42.283,
+# #     latmin=-5.202,
+# #     latmax=6.002,
+# # )
+# ea_region = Region(
+#     name='ea_region',
+#     lonmin=21,
+#     lonmax=51.8,
+#     latmin=-11,
+#     latmax=23,
+# )
+#
+# from src.eng_utils import select_bounding_box_xarray
+
+def select_kenya_region(ds):
+    """ simple helper function to return a subset xarray object for Kenya"""
+    kenya_region = Region(
+        name='kenya',
+        lonmin=33.501,
+        lonmax=42.283,
+        latmin=-5.202,
+        latmax=6.002,
+    )
+    return select_bounding_box_xarray(ds, kenya_region)
+
+
+# ------------------------------------------------------------------------------
+# 1. Wrap all into one function
+# ------------------------------------------------------------------------------
+
+def preprocess_VHI_data(netcdf_filepath, output_dir):
+    """
+    Process:
+    -------
+    * assign time stamp
+    * assign lat lon
+    * create new dataset with these dimensions
+    * Save the output file to new folder
+    """
+    print(f"** Starting work on {netcdf_filepath.split('/')[-1]} **")
+    # 1. read in the dataset
+    ds = xr.open_dataset(netcdf_filepath)
+
+    # 2. extract the timestamp for that file (from the filepath)
+    timestamp = extract_timestamp(ds, netcdf_filepath, use_filepath=True)
+
+    # 3. extract the lat/lon vectors
+    longitudes, latitudes = create_lat_lon_vectors(ds)
+
+    # 4. create new dataset with these dimensions
+    new_ds = create_new_dataset(ds, longitudes, latitudes, timestamp)
+
+    # 5. chop out EastAfrica
+    kenya_ds = select_kenya_region(new_ds)
+
+    # 5. create the filepath and save to that location
+    filename = create_filename(timestamp, netcdf_filepath, subset=True, subset_name="kenya")
+    print(f"Saving to {output_dir}/{filename}")
+    # TODO: change to pathlib.Path objects
+    kenya_ds.to_netcdf(f"{output_dir}/{filename}")
+
+    print(f"** Done for VHI {netcdf_filepath.split('/')[-1]} **")
+    return
+
+
+# ------------------------------------------------------------------------------
+# 1. Run for multiple files
+# ------------------------------------------------------------------------------
+
+def add_coordinates_to_multiple_files(netcdf_filepaths, out_file_dir):
```

Comment:
this is a function to run SEQUENTIALLY for a list of files. Currently being used in `clean_vhi_parallel` to try again with the failed netcdf files